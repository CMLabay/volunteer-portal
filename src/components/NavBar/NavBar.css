.app-nav{
    margin:0;
    background-color:#197fe6;
    height:65px;
}
*{
    font-family:sans-serif;
}
.nav-list{
    padding-top:15px;
    padding-bottom:5px;
    text-align:right;
    margin-top:0;
}
.nav-item{
    display:inline;
}
div{
    margin:0;
}
