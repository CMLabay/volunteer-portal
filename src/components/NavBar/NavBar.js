import React, { Component } from 'react'
import './NavBar.css'
import NavLink from '../NavLink/NavLink'

export default class NavBar extends Component{
    static defaultProps = {
        history: {
          push: () => {},
        },
      }
      render(){
          return(
              <div>
                <nav className='app-nav'>
                    <ul className="nav-list">
                      {Object.keys(this.props.navLinks)
                        .map(key => {
                          return (
                            <NavLink
                              className="nav-item"
                              key={key}
                              id={key}
                              linkName={this.props.navLinks[key].link}
                              routeName={this.props.navLinks[key].route}></NavLink>
                          )
                        })}
                    </ul>
                </nav>
              </div>
          )
      }
    }