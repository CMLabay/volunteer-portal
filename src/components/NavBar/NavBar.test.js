import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import NavBar from './NavBar'
import renderer from 'react-test-renderer';

describe('NavBar component', () => {
    let navLinks = [
        {
            link:"Home", 
            route:"admin"
        },
        {
            link:"Logout",
            route:"/"
        }
    ]
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <NavBar navLinks={navLinks}></NavBar>
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders UI as expected', ()=> {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <NavBar navLinks={navLinks}></NavBar>
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})