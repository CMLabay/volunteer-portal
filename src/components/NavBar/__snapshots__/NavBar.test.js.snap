// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders UI as expected 1`] = `
<div>
  <nav
    className="app-nav"
  >
    <ul
      className="nav-list"
    >
      <li
        className="nav-item"
      >
        <a
          href="/admin"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Logout
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
